src/middlewares/url-validator.ts

```diff
@@ -0,0 +1,19 @@
+import { Request, Response, NextFunction } from 'express';
+import { postUrlSchema, updateUrlSchema } from '../validators/url-validator';
+
+export const validatePostUrl = async (req: Request, res: Response, next: NextFunction) => {
+  const { error, value } = postUrlSchema.validate(req.body);
+  if (error) {
+    return res.status(400).send({ message: error.details[0].message });
```

Comment:
When I don't send anything, I get this error response, which does not make sense.

<img width="784" alt="Screenshot 2024-08-16 at 21 24 05" src="https://github.com/user-attachments/assets/5d6185c3-f884-4c35-9c69-5e5022993163">.

You should find a way to format all the error message that Joi returns, not pick the first one. Maybe this.

```ts
const formatJoiError = (error: Joi.ValidationError) => {
  const formattedError: { [key: string]: string } = {};
  // eslint-disable-next-line unicorn/no-array-for-each
  error.details.forEach((detail: Joi.ValidationErrorItem) => {
    formattedError[detail.path.join('.')] = detail.message;
  });
  return formattedError;
};
```

Reply:
Done